src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py

```diff
@@ -504,6 +504,45 @@ def test_indirect_dofmap_apply(fortran_writer):
 
     check_psyir_results(create_arg_list, fortran_writer)
 
+    psyir_args = create_arg_list.psyir_arglist
+    sym_tab = schedule.symbol_table
+    for ref in psyir_args:
+        assert ref.symbol.name in sym_tab
+
+    # Create a dummy LFRic symbol table to simplify creating
+    # standard LFRic types:
+    dummy_sym_tab = LFRicSymbolTable()
+    int_symbol = dummy_sym_tab.find_or_create_integer_symbol("ignore")
+    for i in [0, 1, 5, 6, 7, 8, 9, 10, 11, 12, 13, 16, 17]:
+        assert psyir_args[i].symbol.datatype == int_symbol.datatype
+
+    # Test all 1d real arrays:
+    real_1d = dummy_sym_tab.find_or_create_array("doesnt_matter1dreal", 1,
+                                                 ScalarType.Intrinsic.REAL)
+    for i in [2, 3]:
+        # The datatype of a field  reference is the type of the member
+        # accessed, i.e. it's the 1D real array.
+        assert psyir_args[i].datatype == real_1d.datatype
+
+    # Test all 2d real arrays:
```

Comment:
Far be it from me to be picky (ahem) but this seems to test 3D arrays (and the comment at L531 also talks about 1d real array).

Reply:
Fair point, sorry. I also replaced all xd with xD, in comments, since I know Iva prefers 1D, ... :)